### PR TITLE
Create BaseSpec. Move functionality from BaseVmSpec to BaseSpec.

### PR DESCRIPTION
**Pull request by skschneider**
Related to #668.
Diffbase #672 (will rebase & replace this branch after that PR is in).

### CODE REVIEW

**Comment by skschneider** [comment]
Now also waiting on #678.

**Comment by skschneider** [comment]
Rebased.

File: perfkitbenchmarker/benchmark_spec.py [hunk 1]
```diff
@@ -211,7 +217,11 @@ def ConstructVirtualMachines(self):
```
Comment (ehankland) [inline perfkitbenchmarker/benchmark_spec.py]: I don't think we need the above try/except blocks anymore

Comment (ehankland) [inline perfkitbenchmarker/benchmark_spec.py]: Ah - nevermind - are you planning on switching BaseDiskSpec in a later PR?

**Comment by ehankland** [comment]
LGTM

File: perfkitbenchmarker/benchmark_spec.py (hunk 1 above)
Comment (ehankland) [inline perfkitbenchmarker/benchmark_spec.py]: Oops - missed that PR... my bad

Comment (skschneider) [inline perfkitbenchmarker/benchmark_spec.py]: Yes, disk specs will be a separate PR.

Comment (ehankland) [inline perfkitbenchmarker/benchmark_spec.py]: Missed https://github.com/GoogleCloudPlatform/PerfKitBenchmarker/issues/668
